auto-rollback/main.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	"time"
+)
+
+func rollback(scriptPath string) {
+	cmd, err := exec.Command("/bin/sh", scriptPath, "restart").Output()
+	log.Println(string(cmd))
+	if err != nil {
+		log.Fatalln("Error while executing rollback script", err)
+	}
+}
+
+func getTicker() *time.Ticker {
+	interval, err := time.ParseDuration(*intervalInSeconds)
+	if err != nil {
+		log.Fatalln("Cannot parse duration.")
+	}
+	return time.NewTicker(interval)
+}
+
+func main() {
+	log.Println("Auto-Rollback")
+	loadConfig()
+
+	prevOrdinal := uint64(0)
+	rollbackInProgress := false
+
+	exit := make(chan string)
+
+	for range getTicker().C {
+		if rollbackInProgress {
+			log.Println("Rollback in progress. Skipping check.")
+		} else {
+
+			ordinal, err := fetchLatestOrdinal()
+			if err != nil {
+				log.Println("Coundn't fetch latest ordinal: %s", err)
+				return
+			}
+
+			if ordinal > prevOrdinal {
+				log.Println("Cluster healthy:", ordinal, ">", prevOrdinal)
+				rollbackInProgress = false
+				prevOrdinal = ordinal
+			} else {
+				log.Println("Rollback needed:", ordinal, "<=", prevOrdinal)
+				rollbackInProgress = true
+				rollback(*rollbackScriptPath)
+
+				time.Sleep(time.Second * 10) // TODO: Remove?
```

Comment:
Is it a check after the rollback script? nodes should be already in Ready state (rollback verifies that to join the nodes), so IMO there is no need for this sleep, but we can try with this right now

Reply:
Sure, let's leave it for now